blockchain/blockchain.go

```diff
@@ -42,6 +47,8 @@ type Reader interface {
 	EventFilter(from *felt.Felt, keys [][]felt.Felt) (*EventFilter, error)
 
 	Pending() (Pending, error)
+
+	SubscribeNewHeads() HeaderSubscription
```

Comment:
I would rather `blockchain` have an `EventListener` interface since subscriptions are probably only useful for RPC. But no need to bother changing it at the moment.

Reply:
> I would rather blockchain have an EventListener interface since subscriptions are probably only useful for RPC.

Would the `blockchain.Reader` have a `SetListener(EventListener)` method? The `rpc.Handler` only accepts a `blockchain.Reader`, not a `blockchain.Blockchain`.

Reply:
If I were to do it, I would implement `blockchain.EventListener` interface in `rpc.Handler` and set it as the event listener.

Reply:
That sounds nice.

> set it as the event listener

If I'm understanding correctly, we would still need to modify the `blockchain.Reader` interface with a `SetListener` method. That would overwrite the existing listener. Is that a fair trade-off?

Reply:
> set it as the event listener.

We can do this in `node` package to avoid modifying `blockchain.Reader`